enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,621 @@
+---
+title: microshift-router-configuration
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@eslutsky"
+  - "@copejon"
+  - "@ggiguash"
+  - "@pmtk"
+  - "@pliurh"
+approvers:
+  - "@dhellmann"
+  - "@jerpeter1"
+api-approvers:
+  - None
+creation-date: 2024-01-08
+last-updated: 2024-01-24
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-2089
+---
+
+# MicroShift router default configuration options
+
+## Summary
+MicroShift's default router is created as part of the platform, but does not
+allow configuring any of its specific parameters. For example, you can not
+disable the router or change its listening ports.
+
+In order to allow these operations and many more, a set of configuration options
+is proposed.
+
+## Motivation
+MicroShift's default ingress router comes as part of the product and is always
+deployed. As part of the start procedure, MicroShift will configure the router
+with fixed parameters, such as being always enabled, and tied to ports 80 and
+443. None of these are configurable for the user, and they also tie to how the
+router is exposed in the cluster.
+
+In order to allow more flexibility, the following configuration parameters shall
+be added:
+- Enable/Disable. Some use cases may be egress only, meaning they do not need
+  an ingress controller. In this case the router pod should not be scheduled
+  to save resources. There should be no rules associated to firewalld or
+  iptables either for an improved security posture.
+- Listening ports. Allow configuring ports other than 80 and 443 improves
+  flexibility.
+- Which IPs the router listens on. Some use cases may require the router to be
+  reachable only from certain networks.
+
+### User Stories
+As a MicroShift user, I want to enable/disable the default router.
+
+As a MicroShift user, I want to disable the default router.
+
+As a MicroShift user, I want to be able to configure in which ports the default
+router is listening.
+
+As a MicroShift user, I want to be able to choose the IPs where the router is
+listening.
+
+As a MicroShift user, I want to be able to allow traffic to the router from
+specific IP addresses.
+
+As a MicroShift user, I want to be able to deny traffic to the router from
+specific IP addresses.
+
+### Goals
+* Allow users to enable the router.
+* Allow users to disable the router.
+* Allow users to reach the router in the configured ports.
+* Allow users to configure in which IPs the router listens.
+* Allow users to allow traffic from specific IP addresses.
+* Allow users to deny traffic from specific IP addresses.
+
+### Non-Goals
+* N/A
+
+## Proposal
```

Comment:
This section is very long, and seems to mix background information and the proposed changes. See if you can summarize the proposal here and move the longer description below to the "Design Details" section.